nx/lib/nx/binary_backend.ex

```diff
@@ -1721,11 +1721,11 @@ defmodule Nx.BinaryBackend do
             a <= b
           end
 
-        fun ->
+        _ ->
           fn a, b ->
             a = binary_to_number(a, type)
             b = binary_to_number(b, type)
-            to_scalar(fun.(a, b)) != 0
+            to_scalar(comparator.(a, b)) != 0
```

Comment:
Given we get this normalized `comparator` function, would it make sense to always use that? Or are the specific `:asc`/`:desc` implementations above more efficient here?

Reply:
Correct, they are quite more efficient because they don't create tensors.